packages/realm/src/Realm.ts

```diff
@@ -1360,6 +1360,10 @@ export namespace Realm {
   export import PushClient = internal.PushClient;
 }
 
-//Set default logger and log level.
+// Clear the internal state to prevent crashes when reloading on React Native
+binding.RealmCoordinator.clearAllCaches();
+binding.App.clearCachedApps();
```

Comment:
Interesting approach 👍 I suppose there's a time between the runtime being torn down until this code runs, where there could potentially fire an event which would crash the app 🤔 But ... it's probably a lot better than nothing.

I wonder if it would make sense to simply call the `Realm.shutdown` instead? That would also cancel outstanding async open tasks, etc.

Also, since this is relevant only for React Native, it might make sense to move it into the platform specific wrapper:
https://github.com/realm/realm-js/blob/5dab79824941ccbb5827603eedcf25537117d5ac/packages/realm/src/platform/react-native/index.ts#L25-L26

Reply:
> Interesting approach

No, just a hack 😄 

> simply call the `Realm.shutdown` instead

I had an issue with `Realm.shutdown` since `RealmProgressPromise` is undefined.

> move it into the platform specific wrapper

I agree.